### PR TITLE
Added deployment features:

### PR DESCRIPTION
**Pull request by crystalhsieh**
Dockerfile for creating deployment image using "make docker_image"
Deployment folder to test that the Docker image will work correctly.

Note: when i was testing the deployed image, proxy handler was throwing an exception when i went to localhost:2020

### CODE REVIEW

File: deployment/example_files/test.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Server Test</title>
+</head>
+<body>
+<h1>Testing testing 1 2 3...</h1>
+<p>Looks like the server is working!</p>
+</body>
+</html>
```
Comment (ryanvoong) [inline deployment/example_files/test.html]: newline

File: deployment/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM busybox:ubuntu-14.04
+
+WORKDIR /opt/webserver
+COPY . /opt/webserver
+
+EXPOSE 2020:2020
+CMD ["./webserver", "test_config"]
```
Comment (ryanvoong) [inline deployment/Dockerfile]: newline

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+RUN apt-get install -y libboost-all-dev
+RUN apt-get install -y make
+RUN apt-get install -y g++
+
+WORKDIR /opt/webserver
+COPY . /opt/webserver
+
+RUN make clean && make
+
+CMD tar -cf - webserver
```
Comment (ryanvoong) [inline Dockerfile]: newline

**Review by ryanvoong — APPROVED** [review]
looks good

File: deployment/example_files/test.html (hunk 1 above)
Comment (crystalhsieh) [reply deployment/example_files/test.html]: done

File: deployment/Dockerfile (hunk 2 above)
Comment (crystalhsieh) [reply deployment/Dockerfile]: done

File: Dockerfile (hunk 3 above)
Comment (crystalhsieh) [reply Dockerfile]: done

**Pull request merged**